### PR TITLE
Make ivy-publish plugin support CC

### PR DESCRIPTION
**Pull request by abstratt**
Applied to ivy-publish similar patterns we already had for maven-publish.

Issue: #13469

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle test ACC

File: subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publisher/IvyDescriptorFileGenerator.java [hunk 1]
```diff
@@ -64,76 +66,111 @@ public void execute(XmlProvider xmlProvider) {
         }
     };
```
Comment (abstratt) [inline subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publisher/IvyDescriptorFileGenerator.java:68]: Here, I extracted a "Model" to encapsulate all the data we require to generate an Ivy descriptor, and a ModelWriter separately from the generator. One benefit of that is that the design could look a lot like the design for MavenPomFileGenerator, which relied on Maven's own Model/ModelWriter code.

File: subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/PublishToMavenRepository.java [hunk 2]
```diff
@@ -110,53 +106,13 @@ private PublishSpec computeSpec() {
...
-        checkCredentialSafety(repository);
-
         MavenNormalizedPublication normalizedPublication = publicationInternal.asNormalisedPublication();
         return new PublishSpec(
                 RepositorySpec.of(repository),
                 normalizedPublication
         );
     }
 
-    private void checkCredentialSafety(DefaultMavenArtifactRepository repository) {
```
Comment (abstratt) [inline subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/PublishToMavenRepository.java:123]: Deleted much of the dead code that I left behind from the "unsafe credentials" kerfuffle, in order to minimize the delta to PublishToIvyRepository.

**Comment by bamboo** [comment]
@bot-gradle test ACC

**Comment by bamboo** [comment]
@bot-gradle test RFN

**Comment by bamboo** [comment]
@bot-gradle test RFN

**Comment by abstratt** [comment]
@bot-gradle test RFN

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Comment by bamboo** [comment]
@bot-gradle test ACC

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Comment by bamboo** [comment]
NOTE FOR REVIEWERS: We intend to come back and rework some of these changes, particularly the ones made to the samples.

**Pull request merged**